File: 1.1/src/c/cz/mg/vulkan/PFN_vkCreateDescriptorSetLayout.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCreateDescriptorSetLayout_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pCreateInfo, jlong pAllocator, jlong pSetLayout, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkCreateDescriptorSetLayout f = (PFN_vkCreateDescriptorSetLayout)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkDescriptorSetLayoutCreateInfo*)jniLongToPointer(pCreateInfo)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator)),
        ((VkDescriptorSetLayout*)jniLongToPointer(pSetLayout))
    );
}


jint Java_cz_mg_vulkan_PFNvkCreateDescriptorSetLayout_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pCreateInfo, jlong pAllocator, jlong pSetLayout){
    (void)env;
    (void)jc;
    PFN_vkCreateDescriptorSetLayout f = (PFN_vkCreateDescriptorSetLayout)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkDescriptorSetLayoutCreateInfo*)jniLongToPointer(pCreateInfo)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator)),
        ((VkDescriptorSetLayout*)jniLongToPointer(pSetLayout))
    );
}

File: 1.1/src/cz/mg/vulkan/PFNvkCreateDescriptorSetLayout.java
package cz.mg.vulkan;

public class PFNvkCreateDescriptorSetLayout extends VkFunctionPointer {
    public PFNvkCreateDescriptorSetLayout() {
    }

    protected PFNvkCreateDescriptorSetLayout(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCreateDescriptorSetLayout(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCreateDescriptorSetLayout(long value) {
        setValue(value);
    }

    public PFNvkCreateDescriptorSetLayout(VkInstance instance) {
        super(instance, new VkString("vkCreateDescriptorSetLayout"));
    }

    public int call(VkDevice device, VkDescriptorSetLayoutCreateInfo pCreateInfo, VkAllocationCallbacks pAllocator, VkDescriptorSetLayout pSetLayout){
        return callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pCreateInfo != null ? pCreateInfo.getVkAddress() : VkPointer.NULL, pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL, pSetLayout != null ? pSetLayout.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callNative(long vkaddress, long device, long pCreateInfo, long pAllocator, long pSetLayout);

}
